.github/workflows/ci.yml

```diff
@@ -48,3 +48,18 @@ jobs:
 
       - name: gotestsum
         run: gotestsum --format short-verbose ./...
+
+      - name: compile for windows
```

Comment:
Would you see any advantage running each stage in parallel, opposed to running them in serie? Something like moving the GOOS definition to the GH action strategy block.

Besides that, I'd like to not run the tests if the build process fails, so I'd move the `go build` stage/s prior to the test one. wdyt?

Reply:
I agree, those are good ideas.  I will:
* add a 3rd dimension to the matrix, called `os` and this can contain the three operating system values "linux", "darwin", and "windows".
* move the compile steps to before the test